pallets/foreign-investments/src/lib.rs

```diff
@@ -263,4 +268,31 @@ pub mod pallet {
 		/// The decrease is greater than the current investment/redemption
 		TooMuchDecrease,
 	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		// The swap is created and now is wating to be fulfilled
+		SwapCreated {
+			who: T::AccountId,
+			swap_id: SwapId<T>,
+			swap: SwapOf<T>,
+		},
+		// The swap was fulfilled by another participant.
+		SwapFullfilled {
+			who: T::AccountId,
+			swap_id: SwapId<T>,
+			remaining: SwapOf<T>,
+			swapped_in: T::SwapBalance,
+			swapped_out: T::SwapBalance,
+		},
+		// The swap was fulfilled by cancelling an opposite swap for the same foreign investment.
+		SwapCancelled {
+			who: T::AccountId,
+			swap_id: SwapId<T>,
+			remaining: SwapOf<T>,
+			cancelled_in: T::SwapBalance,
+			opposite_in: T::SwapBalance,
+		},
+	}
```

Comment:
@hieronx I've needed to add a new `SwapCancelled`. All events are seen from the perspective of who did the swap. 

> Wouldn’t it possible to add an optional arg to the decrease fulfilled event, with the swap ID that was used to cancel it out?

This can not be possible due that swap ID is the same for both directions. Note that swap id is not the order id, which is unknown from FI perspective. As a workaround, I've added the `opposite_in` field to `SwapCancelled,` which has the amount in the opposite direction that has not been canceled. That way, if opposite_in is 0, that means that the cancellation has also canceled the opposite order. That is what you need to know.

Reply:
Looks good to me! Only one nit: in `SwapFulfilled`, wouldn't it make more sense to have the `swapped_out` param before the `swapped_in` param?

Reply:
What do you mean by "before" in this context? Placed in swapped line numbers? Probably, it makes more sense. But right now, we have such conventions in all places in the codebase. I'd prefer to keep it as it is everywhere. But we should probably look into it soon 👍🏻

Reply:
Well we should never change these event definitions again after merging this, at least not without a very strong reason ;) Would be a pain on the subquery side. But sure it is fine to keep it as is then if that is the convention now